### PR TITLE
SDK Framework changes and tests support Listener and SDK both

### PR DESCRIPTION
**Pull request by shresthichauhan**
1. SDK framework changes to support negatives tests.
2. Tests for worker APIs and work order.
3. Jira already raised for failed tests.

Signed-off-by: Shresthi Chauhan <shresthix.chauhan@intel.com>

### CODE REVIEW

File: tests/automation_framework/tests/worker_tests/test_lookup.py [hunk 1]
```diff
@@ -134,3 +134,31 @@ def test_worker_lookup_jsonrpc_different_version(self):
...
+    @pytest.mark.worker_lookup
+    @pytest.mark.test_worker_lookup_diff_unit_length
+    @pytest.mark.listener
+    @pytest.mark.sdk
+    def test_worker_lookup_diff_unit_length(self):
+        request_file = os.path.join(
+            constants.worker_input_file,
+            "worker_lookup_diff_unit_length.json")
+
+        err_cd = self.test_obj.setup_and_build_request_lookup(
```
Comment (jinengandhi-intel) [inline tests/automation_framework/tests/worker_tests/test_lookup.py:149]: Should this test be passing? Can you print the response you get when worker_type 11111 is provided?

**Review by jinengandhi-intel — COMMENTED** [review]
Rest of the tests look okay, I will run it once to check the results.

File: tests/automation_framework/tests/worker_tests/test_update.py [hunk 2]
```diff
@@ -108,3 +108,29 @@ def test_worker_update_invalid_parameter(self):
 
         logger.info('\t\t!!! Test completed !!!\n\n')
 
+    @pytest.mark.worker
```
Comment (jinengandhi-intel) [inline tests/automation_framework/tests/worker_tests/test_update.py:111]: Add this to the bug that I have filed on worker update cases not able to handle negative scenarios: https://jira.devtools.intel.com/browse/TCFH-788

File: tests/automation_framework/tests/worker_tests/test_lookup.py [hunk 3]
```diff
@@ -134,3 +134,31 @@ def test_worker_lookup_jsonrpc_different_version(self):
                 is TestStep.SUCCESS.value)
 
         logger.info('\t\t!!! Test completed !!!\n\n')
+
+
+    @pytest.mark.worker
+    @pytest.mark.worker_lookup
+    @pytest.mark.test_worker_lookup_diff_unit_length
+    @pytest.mark.listener
+    @pytest.mark.sdk
```
Comment (jinengandhi-intel) [inline tests/automation_framework/tests/worker_tests/test_lookup.py:143]: This test case won't be applicable to SDK mode as lookup function takes only values from worker type enum.

File: tests/automation_framework/tests/worker_tests/test_lookup.py (hunk 1 above)
Comment (shresthichauhan) [reply tests/automation_framework/tests/worker_tests/test_lookup.py:149]: Yes test is passing. Response in listener code flow :
[10:46:09 INFO    src.utilities.submit_request_utility] **********Received Request*********
{"jsonrpc": "2.0", "method": "WorkerLookUp", "id": 3, "params": {"workerType": 1111111}}

[10:46:09 INFO    src.utilities.submit_request_utility] **********Received Response*********
{'result': {'totalCount': 0, 'lookupTag': '', 'ids': []}, 'id': 3, 'jsonrpc': '2.0'} 

Notice that totalCount is 0, lookupTag and ids, both are empty for invalid workerType. In contrast, with  workerType 1, receiving response -

{"jsonrpc": "2.0", "method": "WorkerLookUp", "id": 3, "params": {"workerType": 1}}

[10:52:02 INFO    src.utilities.submit_request_utility] **********Received Response*********
{'result': {'totalCount': 1, 'lookupTag': 'ce99e54310adf6d170d6197ac3032f64bcbfa8d583911ba38a96ee60e7926c07', 'ids': ['ce99e54310adf6d170d6197ac3032f64bcbfa8d583911ba38a96ee60e7926c07']}, 'id': 3, 'jsonrpc': '2.0'}

File: tests/automation_framework/tests/worker_tests/test_update.py (hunk 2 above)
Comment (shresthichauhan) [reply tests/automation_framework/tests/worker_tests/test_update.py:111]: Done

**Pull request merged**